torch/_inductor/scheduler.py

```diff
@@ -728,7 +744,8 @@ class FusedSchedulerNode(BaseSchedulerNode):
     @classmethod
     def fuse(cls, node1: BaseSchedulerNode, node2: BaseSchedulerNode):
         assert node1.scheduler is node2.scheduler
-        return cls(node1.scheduler, node1.get_nodes() + node2.get_nodes())
+        assert isinstance(node1, SchedulerNode) and isinstance(node2, SchedulerNode)
+        return cls(node1.scheduler, node1.get_nodes() + node2.get_nodes())  # type: ignore[arg-type]
```

Comment:
just wondering, why is `type: ignore[arg-type]` needed here?

Reply:
`FusedSchedulerNode` expects `List[SchedulerNode]` while `get_nodes()` returns `List[BaseSchedulerNode]`.

Reply:
I see. Looks like `FusedSchedulerNode` and `ForeachKernelSchedulerNode` share the same constructor. Given that it seems to be raising type errors in multiple places (see my comment below re `ForeachKernelSchedulerNode`), maybe it's not being correctly typed to begin with...